src/connector.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+const net = require('net');
+const lookupAll = require('dns-lookup-all');
+
+class Connector {
+  constructor(options, multiSubnetFailover) {
+    this.options = options;
+    this.multiSubnetFailover = multiSubnetFailover;
+  }
+
+  execute(cb) {
+    if (net.isIP(this.options.host)) {
+      this.executeForIP(cb);
+    } else {
+      this.executeForHostname(cb);
+    }
+  }
+
+  executeForIP(cb) {
+    const socket = net.connect(this.options);
+
+    const onError = (err) => {
+      socket.removeListener('error', onError);
+      socket.removeListener('connect', onConnect);
+
+      socket.destroy();
+
+      cb(err);
+    };
+
+    const onConnect = () => {
+      socket.removeListener('error', onError);
+      socket.removeListener('connect', onConnect);
+
+      cb(null, socket);
+    };
+
+    socket.on('error', onError);
+    socket.on('connect', onConnect);
+  }
+
+  executeForHostname(cb) {
+    lookupAll(this.options.host, 4, (err, addresses) => {
```

Comment:
Right. What I mean is that we should default to IPv4, and allow users to specify IPv6 if they want to use it.

If we don't have a default, the DNS lookup will potentially return 2 entries (one with IPv4 and one with IPv6) for each server. When we do serial or parallel connections, we would connect to each server twice - which is weird and unexpected.

I can't imagine there are users with a setup where some servers in their HA setup speak only IPv4 and others speak only IPv6, because that would mean their servers could not even talk to each other.